compiler/ir/backend.jvm/src/org/jetbrains/kotlin/backend/jvm/lower/BridgeLowering.kt

```diff
@@ -127,19 +129,34 @@ private class BridgeLowering(val context: JvmBackendContext) : FileLoweringPass,
         val overriddenSymbols: MutableList<IrSimpleFunctionSymbol> = mutableListOf()
     )
 
-    // Represents a special bridge to `overridden`. Special bridges are always public and non-synthetic.
-    // There are ten type-safe wrappers for different collection methods, which have an additional `methodInfo`
-    // field. There is only one special bridge method which uses generic types in Java (MutableList.removeAt)
-    // and needs a `specializedReturnType` for its overrides. Finally, we sometimes need to use INVOKESPECIAL
-    // to invoke an existing special bridge implementation in a superclass, which is what `superQualifierSymbol`
-    // is for.
+    // Represents a special bridge to `overridden`. Special bridges are overrides for Java methods which are
+    // exposed to Kotlin with a different name or different types. Typically, the Java version of a method is
+    // non-generic, while Kotlin exposes a generic method. In this case, the bridge method needs to perform
+    // additional type checking at runtime. The behavior in case of type errors is configured in `methodInfo`.
+    //
+    // Since special bridges may be exposed to Java as non-synthetic methods, we need correct generic signatures.
+    // There are a total of seven generic special bridge methods (Map.getOrDefault, Map.get, MutableMap.remove with
+    // only one argument, Map.keys, Map.values, Map.entries, and MutableList.removeAt). Of these seven there is only
+    // one which the JVM backend currently handles correctly (MutableList.removeAt). For the rest, it's impossible
+    // to reproduce the behavior of the JVM backend in this lowering.
+    //
+    // Finally, we sometimes need to use INVOKESPECIAL to invoke an existing special bridge implementation in a
+    // superclass, which is what `superQualifierSymbol` is for.
     private data class SpecialBridge(
         val overridden: IrSimpleFunction,
         val signature: Method,
-        val specializedReturnType: IrType? = null,
+        // Does the underlying Java method contains generic type parameters.
```

Comment:
This comment could be interpreted as if the Java method declares its own generic type parameters, which is not the case here (e.g. `Map.getOrDefault` does not declare new type parameters, it uses `Map`'s `V`). Maybe also use more abstract concepts such as "needs generic signature"?

Reply:
I've expanded the comment and changed the field name to avoid potential confusion.